File: nysol_python/mcmd/methods/convTex/db/smp/func_regexrep.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='regexrep'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,caabaa
2,acabaaa
3,
4,cbcbcc
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``id=1,id=2`` の$str$項目にマッチした部分文字列を ``MMM`` に置換する。
'''
script['sh_code']='''
mcal c='regexrep($s{str},"c.*aa","MMM")' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='regexrep($s{str},"c.*aa","MMM")', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/jp/examples/func_regexrep.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,caabaa
2,acabaaa
3,
4,cbcbcc
EOF
)}

############## 例1
title="基本例"
comment=<<'EOF'
\verb|id=1,id=2|の$str$項目にマッチした部分文字列を\verb|MMM|に置換する。
EOF
scp=<<'EOF'
more dat1.csv
mcal c='regexrep($s{str},"c.*aa","MMM")' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

